### PR TITLE
Add standard `Resource > Resource Details` breadcrumb to resource det…

### PR DESCRIPTION
**Pull request by rhamilto**
…ails pages

Fixes https://jira.coreos.com/browse/CONSOLE-1539

Screen Shot 2019-06-14 at 3 20 22 PM

A few special cases:

Screen Shot 2019-06-14 at 3 18 06 PM
Screen Shot 2019-06-14 at 3 18 34 PM
Screen Shot 2019-06-14 at 3 19 14 PM
Screen Shot 2019-06-14 at 3 19 20 PM

Opted not to special case pages where we aggregate resources in to a special list (e.g., Cluster Settings > Global Configuration) to keep things as simple as possible, but can be convinced otherwise.

### CODE REVIEW

**Comment by spadgett** [comment]
@beanh66 @alimobrem

**Comment by spadgett** [comment]
> +25 −97

removing LOC while adding functionality 👍

File: frontend/public/components/factory/details.tsx [hunk 1]
```diff
@@ -55,6 +58,7 @@ export type DetailsPageProps = {
   pages?: Page[];
   pagesFor?: (obj: K8sResourceKind) => Page[];
   kind: K8sResourceKindReference;
+  kindObj?: K8sKind;
```
Comment (spadgett) [inline frontend/public/components/factory/details.tsx]: Do we need both `kind` and `kindObj`?

It looks like we always have a `kindObj` if I'm reading the code below correctly.

**Comment by spadgett** [comment]
@vojtechszocs @christianvogt @serenamarie125 fyi

File: frontend/public/components/factory/details.tsx (hunk 1 above)
Comment (rhamilto) [reply frontend/public/components/factory/details.tsx]: Follow on to do:  update to only pass `kindObj`.

**Comment by rhamilto** [comment]
@spadgett, OLM e2e tests should be fixed.

**Pull request merged**